source/crud/crud.rst

```diff
@@ -236,17 +237,17 @@ Read
      * If specified, drivers SHOULD apply this option to both the original aggregate command and subsequent
      * getMore operations on the cursor.
      *
-     * Drivers MUST NOT specify a batchSize of zero in an aggregate command that includes an $out stage, as
-     * that will prevent the pipeline from executing. Drivers SHOULD leave the cursor.batchSize command option
-     * unset in an aggregate command that includes an $out stage.
+     * Drivers MUST NOT specify a batchSize of zero in an aggregate command that includes an $out or $merge stage,
+     * as that will prevent the pipeline from executing. Drivers SHOULD leave the cursor.batchSize command option
+     * unset in an aggregate command that includes an $out or $merge stage.
```

Comment:
I was originally mistaken in thinking that [SERVER-23923](https://jira.mongodb.org/browse/SERVER-23923) would ensure that the pipeline is executed even if `batchSize` is zero; however, I realized that issue is only ensuring that the `aggregate` command's `writeConcern` is still applied on the `getMore` when iterating such a cursor. Therefore, it's not very relevant to drivers at all given our current design, and this language from [SPEC-826](https://jira.mongodb.org/browse/SPEC-826) must remain in place for 4.2+.

Drivers still need to prohibit sending `batchSize: 0` if they want to guarantee that a pipline with a write stage is executed after a single call to `aggregate()` with no iteration on the return value.

Reply:
Is there any reason to send batchSize with $out/$merge at all? Can we just always ignore it for $out/$merge?

Reply:
Per [SPEC-826](https://jira.mongodb.org/browse/SPEC-826), no driver should be setting `cursor.batchSize` on the initial `aggregate` command when the `batchSize` option is zero. That said, the `batchSize` option may still be relevant for drivers that choose to return a cursor on the output collection; however, we currently don't test for that behavior. If you like, I can add language here that suggests drivers _do_ apply the `batchSize` to any cursor on the output collection.

Testing this would probably require a separate test file, since we'd be covering optional behavior that various drivers would need to skip. If there's interest in that, I'd defer it to someone working on a driver that actually implements the behavior so the spec test isn't written blindly.

Reply:
I'll open a new spec ticket for this discussion. I don't want to hold up this review or expand its scope.

Reply:
Feel free to link it to this issue and SPEC-826 for context.